api/src/api.ts

```diff
@@ -16,36 +37,48 @@ export const plugin = {
   name: "API",
   version: "0.0.2",
   register: async function (server: Hapi.Server, options: ApiOptions) {
-    const { connection: dbConnection } = options;
+    const { connection: dbConnection, config } = options;
 
     server.route({
       method: "GET",
       path: "papers/list",
       handler: async (request) => {
-        let offset;
-        if ('offset' in request.query) {
-          const o = parseInt(
-            Array.isArray(request.query.offset)
-              ? request.query.offset[0]
-              : request.query.offset
-          );
-          if (!isNaN(o)) {
-            offset = o;
+        const offset = firstIntOrDefault(request, "offset", 0);
+        const size = firstIntOrDefault(request, "size", 25);
+
+        const papers = await dbConnection.getAllPapers(offset, size);
```

Comment:
do we need any try/catch-ing around this?

Reply:
Yup, we should. Good catch. I'll add that.

Reply:
So, in this case we're ok actually (I tested what happens).  An error downstream maps to a 500, but doesn't crash the server.

Elsewhere (in this codebase) I've ran into places where code that `throw`s causes the server to crash. Luckily this case isn't one, so we're ok (I think).